CesiumGltfContent/include/CesiumGltfContent/GltfUtilities.h

```diff
@@ -121,5 +125,23 @@ struct CESIUMGLTFCONTENT_API GltfUtilities {
       CesiumGltf::Model& gltf,
       CesiumGltf::Buffer& destination,
       CesiumGltf::Buffer& source);
+
+  /**
+   * @brief Intersects a ray with a glTF model and returns the first
+   * intersection point.
+   *
+   * This function only handles primitives with TRIANGLES or TRIANGLE_FAN mode.
+   * Other modes are ignored.
+   *
+   * @param ray The ray.
+   * @param gltf The glTF model.
+   * @param cullBackFaces An optional boolean flag to indicate whether to cull
+   * backfaces or not. Defaults to true.
+   * @param return The intersection point along the ray, if any.
+   */
+  static std::optional<glm::dvec3> intersectRayGltfModel(
```

Comment:
*Iff* (if and only iff) the barycentric coordinates fall out of the intersection test, that could also be useful. 
(If not, they can be computed post-hoc where necessary, but ... if my dusty memories are not misleading me, I think that this is something that comes "(nearly) for free" for some implementations of ray/triangle intersection tests)

Reply:
The use case that I thought was useful was if you have a hit, and needed traverse back into the model to do more.

For example, I hit something, and have a hit point, but want to know the entire triangle I hit because I want to show it to the user. Similarly, you hit something, and want to show the whole mesh that was hit.

I've made changes to this function to return more info (`HitResult` struct) and modified the tests to check that we can dive into the model with this info after the fact.